File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.py
def solution(n):
    answer = 0  # 변환된 숫자를 저장할 변수

    for _ in range(n):  # n번 반복하여 3x 마을 숫자 찾기
        answer += 1  # 숫자를 하나씩 증가

        # 3의 배수이거나 숫자 3이 포함되면 다음 숫자로 이동
        while answer % 3 == 0 or '3' in str(answer):
            answer += 1

    return answer  # n번째 3x 마을 숫자 반환

# ✅ 예제 테스트 실행
print(solution(15))  # 결과: 25
print(solution(40))  # 결과: 76

File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.java
public class Solution {

    public static int solution(int n) {
        int answer = 0;

        for (int count = 0; count < n; count++) {
            answer++;

            // 3의 배수이거나 숫자에 3이 포함되면 다음 숫자로 넘김
            while (answer % 3 == 0 || String.valueOf(answer).contains("3")) {
                answer++;
            }
        }

        return answer;
    }

    // ✅ 테스트 실행
    public static void main(String[] args) {
        System.out.println(solution(15));  // 출력: 25
        System.out.println(solution(40));  // 출력: 76
    }
}
